sources/httpUtils.ts

```diff
@@ -16,28 +16,28 @@ async function fetch(input: string | URL, init?: RequestInit) {
     input = new URL(input);
 
   let headers = init?.headers;
-  const {username, password} = input;
+
+  const username: string | undefined = input.username ?? process.env.COREPACK_NPM_USERNAME;
+  const password: string | undefined = input.password ?? process.env.COREPACK_NPM_PASSWORD;
+
   if (username || password) {
     headers =  {
       ...headers,
-      authorization: `Bearer ${Buffer.from(`${username}:${password}`).toString(`base64`)}`,
+      authorization: `Basic ${Buffer.from(`${username}:${password}`).toString(`base64`)}`,
     };
+
     input.username = input.password = ``;
-  } else if (input.origin === process.env.COREPACK_NPM_REGISTRY || DEFAULT_NPM_REGISTRY_URL) {
+  }
+
+  if (input.origin === process.env.COREPACK_NPM_REGISTRY || DEFAULT_NPM_REGISTRY_URL) {
```

Comment:
I realize this is not related to this PR, but it looks like the condition would be always true as it is written. We should fix it.
```suggestion
  if (input.origin === process.env.COREPACK_NPM_REGISTRY || input.origin === DEFAULT_NPM_REGISTRY_URL) {
```

Reply:
I will look at this. Thanks for the feedback.